src/plugins/core/sheet.ts

```diff
@@ -1002,12 +1002,18 @@ export class SheetPlugin extends CorePlugin<SheetState> implements SheetState {
   private getImportedSheetSize(data: SheetData): { rowNumber: number; colNumber: number } {
     const positions = Object.keys(data.cells).map(toCartesian);
 
-    let rowNumber = data.rowNumber;
-    let colNumber = data.colNumber;
-    for (let { col, row } of positions) {
-      rowNumber = Math.max(rowNumber, row + 1);
-      colNumber = Math.max(colNumber, col + 1);
+    let rowNumber = data.rowNumber - 1;
+    let colNumber = data.colNumber - 1;
+    for (const { col, row } of positions) {
+      if (row > rowNumber) {
+        rowNumber = row;
+      }
```

Comment:
I tested a bit on the large dataset sheet, and didn't really notice any improvement between master and your PR :x

I really don't think that optimizing the number of "+1" that we do can have any impact on the performances, beside making the code a bit harder to read. 

I guess using `>` rather than Math.max could be slighthly faster, but again, I don't see a difference in practice.

I think we could optimize the function a bit by not looping twice on each cell (on in the main loop, and one in the `map` above). Maybe we loop thrice depending on how Object.keys is implemented ? 

What do you think ?

Reply:
I agree with @hokolomopo here. A loop like 
```typescript
    for (const xc in data.cells) {
      const { col, row } = toCartesian(xc)
``` 
could avoid a guarantee of double allocation of an array of size `data.cells`. I don't know if it's a meaningful difference though...

Reply:
My bad, it was totally obvious...

Concerning the measurements, I have too large a standard deviation with my device to be able to observe a safe difference with `makeLargeDataset`
Here I observe a difference only by measuring isolated code.

Trying to save time on already low percentages is complicated to measure. 
But on several small percentages it may be worth it